### ISSUE TITLE
How to expose embedded items - in auto-generated schemas from Avro-tools

### ISSUE THREAD

**Issue by jeeftor** [kun-fang/avro-data-model#1]
I'm trying to figure out how to work with nested objects that may be defined in an `avdl` file.

So if i take this file `orders.avdl`

```
@namespace("org.jeeftor.avro")
protocol TacoRequest {

enum MeatType{
  CHICKEN,
  BEEF,
  TURKEY,
  FISH
}

enum CheeseType {
  GROSS_VEGAN,
  ACTUAL_COW_CHEESE,
  GOAT_CHEESE
}

enum Toppings {
  LECHUGA,
  TOMATO,
  SAUCE
}

record Taco {
  MeatType meat;
  CheeseType cheese;
  array<Toppings> toppings;
}

record Order {
  union { string, int } order_id;
  array<Taco> tacos;
}

}
```

And convert it to schema with: `avro-tools idl2schemata order.avdl ./parsed`

it will generate the following files:

```
CheeseType.avsc
MeatType.avsc
Order.avsc
Taco.avsc
Toppings.avsc
```
So in short an **Order** is made up of **Taco**s which contain **MeatType** a **CheeseType** and a set of **Toppings**.

I'm wondering if its possible to get "sub-classes" out of this.  In your example you use external references to other records - which DOES work - but I'm trying to automate a process that starts with an `avdl` file which when generating code does not add the references:

So if i run this code:

```python
import datetime
import os

from avro_models import avro_schema, AvroModelContainer

EXAMPLE_NAMES = AvroModelContainer(default_namespace="org.jeeftor.avro")
DIRNAME = os.path.dirname(os.path.realpath(__file__))

@avro_schema(
    EXAMPLE_NAMES,
    schema_file=os.path.join(DIRNAME, "../parsed/Order.avsc"))
class Order(object):
    pass

@avro_schema(
    EXAMPLE_NAMES,
    schema_file=os.path.join(DIRNAME, "../parsed/CheeseType.avsc"))
class CheeseType(object):
    pass
```

I get the following error:

> Traceback (most recent call last):
>   File "python/models.py", line 21, in <module>
>     class CheeseType(object):
>   File "/Users/jstein/.virtualenvs/avro-stack-o_AuMqmg/lib/python3.7/site-packages/avro_models/core.py", line 49, in wrapper
>     schema = SchemaFromJSONData(schema_json, container)
>   File "/Users/jstein/.virtualenvs/avro-stack-o_AuMqmg/lib/python3.7/site-packages/avro/schema.py", line 1214, in SchemaFromJSONData
>     return parser(json_data, names=names)
>   File "/Users/jstein/.virtualenvs/avro-stack-o_AuMqmg/lib/python3.7/site-packages/avro/schema.py", line 1127, in _SchemaFromJSONObject
>     return EnumSchema(name, namespace, symbols, names, doc, other_props)
>   File "/Users/jstein/.virtualenvs/avro-stack-o_AuMqmg/lib/python3.7/site-packages/avro/schema.py", line 712, in __init__
>     other_props=other_props,
>   File "/Users/jstein/.virtualenvs/avro-stack-o_AuMqmg/lib/python3.7/site-packages/avro/schema.py", line 439, in __init__
>     names.Register(self)
>   File "/Users/jstein/.virtualenvs/avro-stack-o_AuMqmg/lib/python3.7/site-packages/avro/schema.py", line 402, in Register
>     'Avro name %r already exists.' % schema.fullname)
> avro.schema.SchemaParseException: Avro name 'org.jeeftor.avro.CheeseType' already exists.

`Avro name 'org.jeeftor.avro.CheeseType' already exists.`

So what I'm wondering is - is there a way to handle my situation without having to go in and modify the schemas and modify them to look more like:

```
  {
      "name": "occupation",
      "type": "example.avro.Occupation"
    },
```

Thanks

**Comment by kun-fang** [comment]
This error is raised from native apache avro python library. The reason is that the library is trying to compile all the schemas under one single namespace. When you compile `Taco.avsc` which includes an enum `CheeseType`, the library will register `CheeseType` in the namespace. When you try to compile `CheeseType.avsc`, it determine the name already exists. 

I see what are trying to do. I can try to create a patch which will find out duplicate schemas (exactly the same schemas) and skip register them with native avro library.

**Comment by kun-fang** [comment]
This is resolved in version 0.0.4

**Comment by jeeftor** [comment]
Awesome. Thanks. Will give it a go tomorrow
